### DISCUSSION TITLE
Get all data

### DISCUSSION THREAD

**Discussion by Aissi17** [uriyyo/fastapi-pagination#542] [Q&A] [answered]
How can we modify our pagination  to include an option for the user to retrieve all the data at once (for example send per_page = -1), without pagination, while still allowing them to use pagination if they choose to do so?

**Comment by uriyyo** [comment, marked as answer]
Hi @Aissi17,

Now `fastapi-pagination` supports optional params:
```py
from fastapi import FastAPI
from pydantic import BaseModel

from fastapi_pagination import Page, add_pagination, paginate
from fastapi_pagination.default import OptionalParams

Page = Page.with_params(OptionalParams)

app = FastAPI()
add_pagination(app)


class UserOut(BaseModel):
    name: str
    surname: str


users = [
    UserOut(name="Steve", surname="Rogers"),
    UserOut(name="Tony", surname="Stark"),
    UserOut(name="Bruce", surname="Banner"),
]


@app.get('/users', response_model=Page[UserOut])
async def get_users():
    return paginate(users)


if __name__ == '__main__':
    import uvicorn

    uvicorn.run(app)
```

**Comment by alexandreczg** [comment]
How do you get away from the linting error that comes from this example. 

Possible related to this issue here in [Pyright](https://github.com/microsoft/pyright/issues/5449) which the author claims to not be a bug but poor hinting on the implementation side. 

![image](https://github.com/uriyyo/fastapi-pagination/assets/131292734/ad4d77c3-e425-4c5a-9e5d-99a68c9f975e)
